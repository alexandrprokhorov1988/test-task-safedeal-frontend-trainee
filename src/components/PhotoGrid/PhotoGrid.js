import React from 'react';
import './PhotoGrid.css';
import PropTypes from "prop-types";
import Image from '../../components/Image/Image';

function PhotoGrid({ cards, onImageClick }) {

  return (
    <section className="photo-grid">
      {cards.map((card) => (
        <Image
          key={card.id}
          id={card.id}
          url={card.url}
          onImageClick={onImageClick}
        />
      ))}
    </section>
  );
}

PhotoGrid.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  onImageClick: PropTypes.func.isRequired,
};

export default PhotoGrid;
