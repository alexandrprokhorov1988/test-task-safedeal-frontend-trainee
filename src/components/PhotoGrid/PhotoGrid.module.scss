.photoGrid {
  margin: 0 auto;
  max-width: 728px;
  width: 95%;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(229px, 1fr));
  grid-row-gap: 30px;
  grid-column-gap: 20px;
  padding-top: 30px;
  padding-bottom: 377px;
  box-sizing: border-box;
}
