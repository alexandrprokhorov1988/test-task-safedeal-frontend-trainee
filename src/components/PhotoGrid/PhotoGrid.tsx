import React from 'react';
import { observer } from 'mobx-react-lite';

import style from './PhotoGrid.module.scss';
import PhotoGridImage from './PhotoGridImage';
import Preloader from '../Preloader';
import photoGrid from '../../stores/photoGridStore';

type PhotoGridProps = {
  /**
   * App onOpen.
   */
  onOpen: () => void;
};

/**
 * PhotoGrid component.
 */
const PhotoGrid: React.FC<PhotoGridProps> = observer(({ onOpen }) => {
  console.log('photogrid');// todo del

  React.useEffect(() => {
    console.log('запрос на сервер за карточками'); // todo del
    photoGrid.getInitialImages();
  }, []);

  return (
    <section className={style.photoGrid}>
      {photoGrid.isLoading ? <Preloader />
        : (
          <>
            {photoGrid.cards && photoGrid.cards.map((card: any) => (
              <PhotoGridImage
                key={card.id}
                id={card.id}
                url={card.url}
                onOpen={onOpen}
              />
            ))}
          </>
        )}
    </section>
  );
});

export default React.memo(PhotoGrid);
