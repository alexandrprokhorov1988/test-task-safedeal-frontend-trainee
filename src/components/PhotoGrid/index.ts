export {default} from './PhotoGrid';
export {default as PhotoGridImage} from './PhotoGridImage';
