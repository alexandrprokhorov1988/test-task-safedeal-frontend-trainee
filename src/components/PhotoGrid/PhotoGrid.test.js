import React from 'react';
import {render, unmountComponentAtNode} from "react-dom";
import renderer from 'react-test-renderer';
import {act} from "react-dom/test-utils";
import PhotoGrid from './PhotoGrid';

let container = null;
const fakeCards = [
  {
    id: 237,
    url: "https://picsum.photos/id/237/300/200",
  }
];

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});
afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('PhotoGrid component', () => {
  it("Должен отрисовывать картинки c заданными аттрибутами", async () => {
    act(() => {
      render(<PhotoGrid cards={fakeCards} onImageClick={() => {
      }}/>, container);
    });
    const img = container.querySelector(".photo-grid-img");
    expect(img.getAttribute("src")).toEqual("https://picsum.photos/id/237/300/200");
    expect(img.getAttribute("alt")).toEqual("Картинка");
    expect(img.getAttribute("role")).toEqual("presentation");
    expect(img.getAttribute("tabIndex")).toEqual("0");
  });

  it('Проверка snapshot', () => {
    const render = renderer
      .create(<PhotoGrid cards={fakeCards} onImageClick={() => {
      }}/>)
      .toJSON();
    expect(render).toMatchSnapshot();
  });
});
