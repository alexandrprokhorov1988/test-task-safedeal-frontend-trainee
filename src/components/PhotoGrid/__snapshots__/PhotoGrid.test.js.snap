// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGrid component Проверка snapshot 1`] = `
<section
  className="photo-grid"
>
  <img
    alt="Картинка"
    className="photo-grid-img"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    src="https://picsum.photos/id/237/300/200"
    tabIndex={0}
  />
</section>
`;
